content/automate/ManageIQ/Transformation/Infrastructure/VM/Common.class/__methods__/poweron.rb

```diff
@@ -11,8 +11,10 @@ def initialize(handle = $evm)
 
               def main
                 task = @handle.root['service_template_transformation_plan_task']
-                destination_vm = @handle.vmdb(:vm).find_by(:id => task.get_option(:destination_vm_id))
-                destination_vm.start
+                if task.get_ption(:source_vm_power_state) == 'on'
+                  destination_vm = @handle.vmdb(:vm).find_by(:id => task.get_option(:destination_vm_id))
+                  destination_vm.start
```

Comment:
@fdupont-redhat Do you need to be concerned if the find_by doesn't find the VM?

Reply:
I don't think so as the previous state in the state machine waits for the VM to be in the VMDB.